docs/posts/ibis-bench/index.qmd

```diff
@@ -0,0 +1,1313 @@
+engine to benchmark. We don't because it's not currently a backend for Ibis,
+though [there has been work done to make it
+one](https://github.com/ibis-project/ibis/pull/8497).
+
+If you're interested in contributing to Ibis, a new backend like chDB could be a
+great project for you!
+:::
+
+9/22 queries for Ibis with the Polars backend fail from [lack of scalar subquery
+support](#failing-polars-queries). Due to this and relatively experimental SQL
+support in Polars, we've opted to run on [the Polars API
+directly](https://github.com/lostmygithubaccount/ibis-bench/blob/v2.0.0/src/ibis_bench/queries/polars.py)
+in this iteration of the benchmark. This is done with the LazyFrames API **and
+no streaming engine** (per the Polars team's recommendation, as the current
+streaming engine is being rewritten). This also allows us to compare the
+performance of the Polars backend through Ibis with the Polars API directly for
+the queries that do succeed.
+
+#### Failing queries
+
+Queries fail for one of two reasons:
+
+1. The query doesn't work in the given system
+2. The query otherwise failed on a given run
+
+We'll note the cases of the first below. The second is usually due to memory
+pressure and [will be seen at higher scale
+factors](#failing-queries-due-to-memory-pressure) throughout the data.
+
+#### Failing DataFusion queries
+
+Queries 16, 21, and 22 fail for the DataFusion backend via Ibis dataframe code,
+and query 16 fails through SQL. Note that [all TPC-H SQL queries successfully
+run through DataFusion
+directly](https://github.com/apache/datafusion-benchmarks) -- Ibis generates SQL
+that [hits a bug with DataFusion that has already been
+fixed](https://github.com/apache/datafusion/issues/10830).  We expect these
+queries to work in the next iteration of this benchmark coming soon.
+
+#### Failing Polars queries
+
+Queries 11, 13-17, and 20-22 fail for the Polars backend via Ibis dataframe
+code. These all fail due to lack of scalar subquery support in the backend. I've
+[opened an issue](https://github.com/ibis-project/ibis/issues/9422) for tracking
+and discussion.
+
+:::{.callout-tip title="Interested in contributing?"}
+Interesting in contributing to Ibis? Increasing coverage of operations for a
+backend is a great place to start!
+:::
+
+### How queries are written
+
+See [the source
+code](https://github.com/lostmygithubaccount/ibis-bench/tree/v2.0.0/src/ibis_bench/queries)
+for the exact queries used in this iteration of the benchmark. Polars recently
+updated their TPC-H queries, so the next iteration of this benchmark would use
+those.
+
+Queries were adapted from [Ibis TPC-H
+queries](https://github.com/ibis-project/ibis/tree/main/ibis/backends/tests/tpch)
+and [Polars TPC-H queries](https://github.com/pola-rs/tpch). The first 10 Ibis
+dataframe queries were translated from the Polars dataframe queries, while the
+rest were directly adapted from the Ibis repository. The SQL strings were
```

Comment:
```suggestion
rest were directly adapted from the Ibis repository. The SQL strings were
```

This could link to our test directory if you wanted, but not necessary.

Reply:
it is linked above